File: Two-pointer/20922번 - 겹치는 건 싫어/20922.py
import sys
from collections import defaultdict

input = sys.stdin.readline

def solve(n, k, arr):
    left, seq_len = 0, 0
    cnt = defaultdict(int)
    ans = 0
    
    for right in range(n):
        cnt[arr[right]] += 1
        
        while cnt[arr[right]] > k:
            cnt[arr[left]] -= 1
            
            if cnt[arr[left]] == 0:
                del cnt[arr[left]]
                
            left += 1
        
        ans = max(ans, right - left + 1)
    
    print(ans)

if __name__ == '__main__':
    n, k = map(int, input().split())
    arr = list(map(int, input().split()))
    solve(n, k, arr)

File: Two-pointer/20922번 - 겹치는 건 싫어/20922.cpp
#include <iostream>

using namespace std;

int n, k;
int arr[200001] = { 0, };
int cnt[100001] = { 0, };

void input() {
    cin >> n >> k;
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }
}

void solve() {
    int len = 0, ans = 0;
    
    for(int i = 0, j = 0; i < n; i++) {
        int cur = arr[i];
        cnt[cur]++;
        len++;
        
        while(cnt[cur] > k) {
            cnt[arr[j++]]--;
            len--;
        }
        
        ans = max(ans, len);
    }
    
    cout << ans << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}